src/filters.ts

```diff
@@ -5,61 +5,116 @@ import type {
   IntrospectionType,
 } from 'graphql'
 
-export const mapFilterType = (type: IntrospectionNamedTypeRef, value: any, key: string) => {
-  const normalizedName = type.name.toLowerCase()
-  switch (normalizedName) {
-    case 'boolean':
-      return {
-        [key]: {
-          equalTo: value,
-        },
-      }
-    case 'string':
-      return {
-        or: [
-          {
-            [key]: {
-              equalTo: value,
-            },
-          },
-          {
-            [key]: {
-              like: `%${value}%`,
-            },
-          },
-        ],
-      }
-    case 'uuid':
-    case 'bigint':
-    case 'int':
-      return Array.isArray(value)
-        ? {
-            [key]: {
-              in: value,
-            },
-          }
-        : {
-            [key]: {
-              equalTo: value,
-            },
-          }
-    default:
-      throw new Error(`Filter for type ${type.name} not implemented.`)
+import type {
+  Filter,
+  FilterMap,
+  FilterSpec,
+  Operator
+} from './types'
+
+/**
+ * Transforms for a certain field type a search value for a field (the key)
+ * to a filter that is understood by postgraphile.
+ *
+ * For example:
+ *
+ * - type: {kind: "SCALAR", name: "String", ofType: null, __typename: "__Type"}
+ * - value: "some keyword"
+ * - key: "name"
+ *
+ * Is transformed to:
+ *
+ * ```ts
+ * {
+ *   name: {
+ *     includes: "some keyword"
+ *   }
+ * }
+ * ```
+ */
+export const mapFilterType = (
+  type: IntrospectionNamedTypeRef,
+  value: any,
+  key: string
+): Filter | undefined => {
+  if (Array.isArray(value)) {
+    const spec: FilterSpec = {
+      operator: 'in',
+      value
+    }
+
+    value = spec
+  }
+
+  if (typeof value !== 'object') {
+    const typeName = (type?.name ?? '').toLowerCase()
+
+    let operator: Operator = 'equalTo'
+    // string uses includes a the default operator for historical reasons
+    if (typeName === 'string') {
+      operator = 'includes'
+    }
+
+    // a type of FullText uses matches a the default operator for historical reasons
```

Comment:
I actually want to remove this and require the user to explicitly build this operator.